### PR TITLE
Upload buffer support

### PR DESCRIPTION
**Pull request by RachelFranks**
Adds unmarshalling code in ArbOS and buffer upload code in rust that allows us to support non-empty buffers. This also includes a minimal patch from future PR #568 to enable string support and multi-line comments.

### CODE REVIEW

File: src/compile/codegen.rs [hunk 1]
```diff
@@ -817,12 +817,32 @@ fn mavm_codegen_expr<'a>(
...
         TypeCheckedExprKind::NewBuffer => {
             let opcode = Opcode::AVMOpcode(AVMOpcode::NewBuffer);
             code.push(Instruction::new(opcode, None, debug));
             Ok((label_gen, code, num_locals))
         }
+        TypeCheckedExprKind::Quote(bytes) => {
+            code.push(create!(
+                Noop,
+                Value::new_tuple(vec![
+                    Value::Int(Uint256::from_usize(2 * bytes.len())),
```
Comment (edfelten) [inline src/compile/codegen.rs:840]: Why do we multiply bytes.len by 2 here?

**Review by edfelten — COMMENTED** [review]
One question. Otherwise, looks good.

File: src/compile/codegen.rs (hunk 1 above)
Comment (RachelFranks) [reply src/compile/codegen.rs]: Oops, that's wrong. I've committed a fix

Comment (edfelten) [reply src/compile/codegen.rs]: Thanks.  Looks good.

**Pull request merged**